Add operating_system option

petr-balogh opened the pull request:
Once PR https://github.com/ansible/ansible/pull/38211 gets merged this
patch adds posibility to use this option from image-template role.

petr-balogh committed:
Add params for os and memory

PR https://github.com/ansible/ansible/pull/38211 got merged but will be
available in 2.6.
This patch adds posibility to use these params from image-template role.

machacekondra reviewed (changes requested):
This parameter must be added also to file: `tasks/qcow2_image.yml` and we must create `library` directory with the updated module.

petr-balogh committed:
Add yamlint config file

set line-length rules:
  max: 80
  allow-non-breakable-words: yes
  allow-non-breakable-inline-mappings: yes

petr-balogh commented:
Hi Ondra, comments are done, unfortunately I did force push so you won't see changes with previous code.

machacekondra reviewed (approved):
Thanks!
